x-pack/plugin/versionfield/src/test/java/org/elasticsearch/xpack/versionfield/VersionStringFieldTests.java

```diff
@@ -0,0 +1,498 @@
+            .get();
+        assertEquals(2, response.getHits().getTotalHits().value);
+
+        // ranges excluding edges
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.rangeQuery("version").from("1.0.0", false).to("3.0.0")).get();
+        assertEquals(3, response.getHits().getTotalHits().value);
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.rangeQuery("version").from("1.0.0").to("2.1.0", false)).get();
+        assertEquals(3, response.getHits().getTotalHits().value);
+
+        // open ranges
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.rangeQuery("version").from("1.4.0")).get();
+        assertEquals(4, response.getHits().getTotalHits().value);
+
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.rangeQuery("version").to("1.4.0")).get();
+        assertEquals(2, response.getHits().getTotalHits().value);
+    }
+
+    public void testPrefixQuery() throws IOException {
+        String indexName = setUpIndex("test");
+        // prefix
+        SearchResponse response = client().prepareSearch(indexName).setQuery(QueryBuilders.prefixQuery("version", "1")).get();
+        assertEquals(3, response.getHits().getTotalHits().value);
+
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.prefixQuery("version", "2.1")).get();
+        assertEquals(2, response.getHits().getTotalHits().value);
+
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.prefixQuery("version", "2.1.0-")).get();
+        assertEquals(1, response.getHits().getTotalHits().value);
+
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.prefixQuery("version", "1.3.0+b")).get();
+        assertEquals(1, response.getHits().getTotalHits().value);
+
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.prefixQuery("version", "2")).get();
+        assertEquals(3, response.getHits().getTotalHits().value);
+
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.prefixQuery("version", "21")).get();
+        assertEquals(1, response.getHits().getTotalHits().value);
+
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.prefixQuery("version", "21.")).get();
+        assertEquals(1, response.getHits().getTotalHits().value);
+
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.prefixQuery("version", "21.1")).get();
+        assertEquals(1, response.getHits().getTotalHits().value);
+
+        response = client().prepareSearch(indexName).setQuery(QueryBuilders.prefixQuery("version", "21.11")).get();
+        assertEquals(1, response.getHits().getTotalHits().value);
+    }
+
+    public void testSort() throws IOException {
+        String indexName = setUpIndex("test");
+
+        // sort based on version field
+        SearchResponse response = client().prepareSearch(indexName)
```

Comment:
Do you think we need to add some illegal versions and test that sort ASC will put them at the end?

Reply:
Sure can do.